hub/core/chunk_engine/tests/test_chunk_generator.py

```diff
@@ -0,0 +1,171 @@
+from hub.core.chunk_engine.generator import chunk
+
+from util import (
+    make_dummy_byte_array,
+    get_random_chunk_size,
+    get_random_num_samples,
+    get_random_partial,
+    assert_valid_chunk,
+)
+
+
+TRIALS_PER_TEST = 10
```

Comment:
Randomized testing is bad. If we don't discriminate against the content, there's no need to randomize it. And if execution differs based on the input size, test each of the cases separately for better coverage

Reply:
true, i will use configs instead.

Reply:
I'm not sure what you mean by configs, but in the sake of clarity, here's a **complete** example of a test.

```python
def test_perfect_fit():
    data = b"\x00\xaa\xbb\xff"
    result = list(chunk(data, chunk_size=2))
    assert result[0][0] == b"\x00\xaa"
    assert result[1][0] == b"\xbb\xff"
```

Reply:
right. there are lots of edge cases that need to be accounted for. i am using `pytest.mark.parametrize` now to provide a list of arguments & expected values for these so i can write one function & call it will deterministic sets of parameters.

Reply:
@benchislett so would you say generating a random set of bytes is a bad thing to do? i am designing the tests now to have configs, but the bytes that are being chunked are random (deterministic size).